taskcat/_s3_stage.py

```diff
@@ -27,7 +29,12 @@ def stage_in_s3(buckets, project_name, project_root):
     for test in buckets.values():
         for bucket in test.values():
             distinct_buckets[bucket.name] = bucket
-    for bucket in distinct_buckets.values():
-        S3Sync(
-            bucket.s3_client, bucket.name, project_name, project_root, bucket.object_acl
-        )
+    pool = ThreadPool(32)
+    func = partial(_sync_wrap, project_name=project_name, project_root=project_root)
+    pool.map(func, distinct_buckets.values())
```

Comment:
I see no issues with your changes in this commit. However line 31 is not partition-safe. 

Recommend line 31:
`distinct_buckets[f"{bucket.name}-{bucket.partition}"] = bucket`

Reply:
I'm not following how this is not partition safe, what is the scenario where we'll end up with the same bucket name in multiple partitions?.  (also, this pr does not change line 31, and afaik x-partition upload is working fine atm)

Reply:
Existing buckets. 

- foobar-blah in commerical partition
- foobar-blah in China. 

With the existing L31, one of them will not be uploaded to. Cross-partition uploading works under our existing scenarios because we generate the bucket name (`taskcat-<HASH>-<REGION>`)

In a bring-your-own bucket scenario, it's entirely possible the bucket name is re-used across multiple partitions.

Reply:
yeah, seeing as that's an existing limitation, lets handle elsewhere.